src/app/app.component.ts

```diff
@@ -24,9 +29,56 @@ export class AppComponent implements OnInit {
       .subscribe(response => this.isAngularProject = !!response);
   }
 
+  checkCommandStatus(commandId: string): void {
+    this.commandService.checkCommandStatus(commandId)
+      .subscribe(response => {
+        this.runningCommands[commandId] = response;
+      }, error => {
+        if (error.status === 404) {
+          console.log('command not found in server, stop checking');
+          this.doneCheckingCommand(commandId);
+        }
+      });
+  }
+
+  doneCheckingCommand(commandId: string): void {
+    this.subscription[commandId].unsubscribe();
+    this.subscription[commandId] = null;
+  }
+
+  commandDone(commandId: string): void {
+    this.checkAngularProject();
+    this.runningCommands[commandId] = null;
+  }
+
+  startCheckingCommand(commandId: string): void {
+    if (this.runningCommands[commandId]) {
+      const status = this.runningCommands[commandId].status;
+
+      if (status === 'done') {
+        console.log('done');
+        this.commandDone(commandId);
+        this.doneCheckingCommand(commandId);
```

Comment:
should be above `this.commandDone...`

Reply:
`doneCheckingCommand` is `unsubscribe`ing from the `timedStatusCheck` and by doing so, wouldn't it stop `commandDone` from executing after it?

Reply:
why should it stop it?

Reply:
you can check it very simply

Reply:
I've checked it and it didn't stop but I don't understand why...
shouldn't `unsubscribe()` stop the `Observable` and it's `callback`?